From 6dcfc3daab7c1c459b36261163b4e1dc15a6e965 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 3 Feb 2016 16:12:54 +0000
Subject: [PATCH 141/170] BCM270X_DT: Disable DMA for bcm2835-sdhost on Pi2

This is an interim patch to verify that problems seen with
some cards can be worked around at the expense of an increased
CPU load by forcing PIO mode.

Although this could have used the brcm,force-pio attribute, that
is a boolean (true if present, false if absent) and attributes
can't be deleted by overlays. Instead, make brcm,pio-limit
unfeasibly high instead to allow DMA to be re-enabled using the
pio_limit parameter of the sdhost or sdtweak overlays.
---
 arch/arm/boot/dts/bcm2709-rpi-2-b.dts | 1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
+++ b/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
@@ -43,6 +43,7 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&sdhost_pins>;
 	bus-width = <4>;
+	brcm,pio-limit = <0x7fffffff>;
 	status = "okay";
 };
 
